File: MODUL6/PraktikumC/PRAK604-2310817110004-ABDURRAHMANGILANGHARJUNA.c
#include <stdio.h>
#include <string.h>

void kodepesan(const char *kode, const char *pesan) {
    int panjangKode = strlen(kode);
    int panjangPesan = strlen(pesan);

    if (panjangKode != panjangPesan) {
        printf("Panjang kalimat berbeda, pesan palsu\n");
        return;
    }

    char hasil[panjangPesan + 1];
    int jumlahBintang = 0, jumlahPagar = 0;

    for (int i = 0; i < panjangKode; i++) {
        if (kode[i] == pesan[i]) {
            hasil[i] = '*';
            jumlahBintang++;
        } else {
            hasil[i] = '#';
            jumlahPagar++;
        }
    }

    hasil[panjangPesan] = '\0';
    printf("%s\n", hasil);
    printf("* = %d\n", jumlahBintang);
    printf("# = %d\n", jumlahPagar);

    if (jumlahBintang >= jumlahPagar) {
        printf("Pesan Asli\n");
    } else {
        printf("Pesan Palsu\n");
    }
}

int main() {
    char kode[100], pesan[100];
    fgets(kode, sizeof(kode), stdin);
    kode[strcspn(kode, "\n")] = '\0';

    printf("");
    fgets(pesan, sizeof(pesan), stdin);
    pesan[strcspn(pesan, "\n")] = '\0';

    kodepesan(kode, pesan);

    return 0;
}

File: MODUL6/PraktikumPy/PRAK604-2310817110004-ABDURRAHMANGILANGHARJUNA.py
def kode_pesan(kode, pesan):
    panjang_kode = len(kode)
    panjang_pesan = len(pesan)

    if panjang_kode != panjang_pesan:
        print("Panjang kalimat berbeda, pesan palsu")
        return

    hasil = ""
    jumlah_bintang = 0
    jumlah_pagar = 0

    for i in range(panjang_kode):
        if kode[i] == pesan[i]:
            hasil += '*'
            jumlah_bintang += 1
        else:
            hasil += '#'
            jumlah_pagar += 1

    print(hasil)
    print(f"* = {jumlah_bintang}")
    print(f"# = {jumlah_pagar}")

    if jumlah_bintang >= jumlah_pagar:
        print("Pesan Asli")
    else:
        print("Pesan Palsu")

if __name__ == "__main__":
    kode = input()
    pesan = input()
    kode_pesan(kode, pesan)
